sky/compositor/instrumentation.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_COMPOSITOR_INSTRUMENTATION_H_
+#define SKY_COMPOSITOR_INSTRUMENTATION_H_
+
+#include "base/macros.h"
+#include "base/time/time.h"
+
+namespace sky {
+namespace compositor {
+namespace instrumentation {
+
+class Stopwatch {
+ public:
+  class ScopedLap {
+   public:
+    explicit ScopedLap(Stopwatch& stopwatch) : _stopwatch(stopwatch) {
+      _stopwatch.start();
+    }
+
+    ~ScopedLap() { _stopwatch.stop(); }
+
+   private:
+    Stopwatch& _stopwatch;
+
+    DISALLOW_COPY_AND_ASSIGN(ScopedLap);
+  };
+
+  explicit Stopwatch() : _start(base::TimeTicks::Now()), _lastLap() {}
+
+  const base::TimeDelta& lastLap() const { return _lastLap; }
+
+  base::TimeDelta currentLap() const { return base::TimeTicks::Now() - _start; }
+
+  void start() { _start = base::TimeTicks::Now(); }
+
+  void stop() { _lastLap = base::TimeTicks::Now() - _start; }
+
+ private:
+  base::TimeTicks _start;
+  base::TimeDelta _lastLap;
+
+  DISALLOW_COPY_AND_ASSIGN(Stopwatch);
+};
+
+class Counter {
+ public:
+  explicit Counter() : _count(0) {}
+
+  size_t count() const { return _count; }
+
+  void reset(size_t count = 0) { _count = count; }
+
+  void increment(size_t count = 1) { _count += count; }
+
+ private:
+  size_t _count;
+
+  DISALLOW_COPY_AND_ASSIGN(Counter);
+};
```

Comment:
The idea is that certain counters that measure frame specific (or other "instantaneous") information  are useless on their own and more insights can be gleaned from graphing out how these values change over time. For such cases, the past `n` counts could be recorded (on `reset`) and charted. So this class is more of a stub.